scalameter-core/src/main/scala/org/scalameter/InvocationCountMatcher.scala

```diff
@@ -0,0 +1,123 @@
+package org.scalameter
+
+import java.lang.reflect.Method
+import java.util.regex.Pattern
+import org.objectweb.asm.Type
+import scala.util.matching.Regex
+
+
+sealed trait ClassMatcher {
+  /** Matches class.
+   *
+   *  Note that any '/' in a class name is replaced by a '.'.
+   */
+  def matches(className: String): Boolean
+}
+
+/** Matches class with a class name given as a string. */
+case class StringClassMatcher(clazz: String) extends ClassMatcher {
+  def this(clazz: Class[_]) = this(clazz.getName)
+  
+  def matches(className: String): Boolean = className.replace('/', '.') == clazz
+}
+
+object StringClassMatcher {
+  def apply(clazz: Class[_]) = new StringClassMatcher(clazz)
+}
+
+/** Matches class with a regex. */
+case class RegexClassMatcher(regex: Pattern) extends ClassMatcher {
+  def matches(className: String): Boolean = regex.matcher(className.replace('/', '.')).matches()
+}
+
+sealed trait MethodMatcher {
+  /** Matches method.
+   *
+   *  @param methodName method name
```

Comment:
The meaning of these two parameters is not obvious, so please make sure to precisely document what they mean.